### PR TITLE
LCH-5270: Added new method query entities

### PR DESCRIPTION
**Pull request by ankitsingh0188**

### CODE REVIEW

**Comment by ankitsingh0188** [comment]
Build will fail because we need to merge the https://github.com/acquia/content-hub-php/pull/133 PHP Lib PR first.

File: src/ContentHubClient.php [hunk 1]
```diff
@@ -1284,5 +1285,20 @@ public function cancelScroll(string $scroll_id): ?array {
...
+   * @param array $params
+   *   Query params.
+   *
+   * @return array|null
+   *   Response from backend call.
+   *
+   * @throws \Exception
+   */
+  public function queryEntities(array $params): ?array {
+    return self::getResponseJson($this->get("entities", [RequestOptions::QUERY => $params]));
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: Please use single quotes.

File: src/ContentHubClient.php [hunk 2]
```diff
@@ -1284,5 +1285,20 @@ public function cancelScroll(string $scroll_id): ?array {
 
     return self::getResponseJson($this->delete("scroll", $options));
   }
+  
+  /**
+   * Fetch entities via query params.
```
Comment (saxumVermes) [inline src/ContentHubClient.php]:
```suggestion
   * Fetches entities via query params.
```

File: src/ContentHubClient.php [hunk 3]
```diff
@@ -1284,5 +1285,20 @@ public function cancelScroll(string $scroll_id): ?array {
...
+   *
+   * @param array $params
+   *   Query params.
+   *
+   * @return array|null
+   *   Response from backend call.
+   *
+   * @throws \Exception
+   */
+  public function queryEntities(array $params): ?array {
```
Comment (saxumVermes) [inline src/ContentHubClient.php]: Let's make this parameter optional.

File: src/ContentHubClient.php [hunk 4]
```diff
@@ -1284,5 +1285,20 @@ public function cancelScroll(string $scroll_id): ?array {
 
     return self::getResponseJson($this->delete("scroll", $options));
   }
+  
+  /**
+   * Fetch entities via query params.
+   *
+   * @param array $params
+   *   Query params.
```
Comment (saxumVermes) [inline src/ContentHubClient.php:1290]: Let's include the expected array structure into the documentation like:
```php
 [ 'param1' => 'param_value' ]
```

**Review by saxumVermes — CHANGES_REQUESTED** [review]
We need to follow-up on the change in ObjectFastory::getCHClient() as well.

File: test/ContentHubClientTest.php [hunk 5]
```diff
@@ -2655,4 +2656,27 @@ public function testCancelScrollIfSucceeds(): void {
...
+    ];
+
+    $this->ch_client
+      ->shouldReceive('get')
+      ->once()
+      ->with('entities', [RequestOptions::QUERY => $request_parameters])
+      ->andReturn($this->makeMockResponse(SymfonyResponse::HTTP_OK, [], json_encode($response)));
+
+    $this->assertSame($this->ch_client->queryEntities($request_parameters), $response);
+  }
```
Comment (saxumVermes) [inline test/ContentHubClientTest.php:2680]: After we made the array parameter optional, we need another test case, testing without any argument passed to the `queryEntities` method.

Comment (ankitsingh0188) [reply test/ContentHubClientTest.php:2680]: Done.

File: src/ContentHubClient.php (hunk 3 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php]: Done.

File: src/ContentHubClient.php (hunk 2 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php]: Done.

File: src/ContentHubClient.php (hunk 1 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php]: Done.

**Comment by ankitsingh0188** [comment]
> We need to follow-up on the change in ObjectFastory::getCHClient() as well.

Refactored the ObjectFactory::getCHClient() method and it's usage.

File: src/ContentHubClient.php (hunk 4 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php:1290]: Sure.

File: src/ContentHubClient.php [hunk 6]
```diff
@@ -3,18 +3,15 @@
...
 use Acquia\Hmac\Guzzle\HmacAuthMiddleware;
-use Exception;
 use GuzzleHttp\Client;
 use GuzzleHttp\Exception\BadResponseException;
 use GuzzleHttp\Exception\ClientException;
 use GuzzleHttp\Exception\RequestException;
-use GuzzleHttp\HandlerStack;
 use GuzzleHttp\Psr7\Response;
-use Psr\Log\LogLevel;
+use GuzzleHttp\RequestOptions;
```
Comment (saxumVermes) [inline src/ContentHubClient.php:14]: Thanks for removing the unused imports!

**Review by saxumVermes — APPROVED** [review]
Looks good, let's roll it!

File: src/ContentHubClient.php (hunk 6 above)
Comment (ankitsingh0188) [reply src/ContentHubClient.php:14]: My Pleasure 👍

**Review by amangrover90 — APPROVED** [review]
Looks good to me!!

**Pull request merged**